src/screens/WasteCollectionScreen.js

```diff
@@ -169,7 +173,7 @@ export const WasteCollectionScreen = ({ navigation }) => {
   const filterStreets = useCallback(
     (currentInputValue, addressesData) => {
       if (wasteAddressesTwoStep && inputValueCity === '') return [];
-      if (currentInputValue === '') return [];
+      if (currentInputValue === '' && !isStreetInputFocused && isInputAutoFocus) return [];
```

Comment:
please check the gobal setting first, before checking the values for the logic. if the global setting is false, we do not have to check the other values, right? this is more naturally readable in my eyes.

Reply:
You're talking about this control?
```
if (isInputAutoFocus && currentInputValue === '' && !isStreetInputFocused) return [];
```
 or 
```
if (isInputAutoFocus) return [];
if (currentInputValue === '' && !isStreetInputFocused) return [];
```

What do you mean by globalSettings?

Reply:
> What do you mean by globalSettings?

`isInputAutoFocus `

you understood correctly with `if (isInputAutoFocus && currentInputValue === '' && !isStreetInputFocused) return [];`

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/4e36378f9f2ac4c5bc8c431ed435eddc4c91ecf0